src/System.IO.FileSystem/tests/File/Create_str.cs

```diff
@@ -0,0 +1,231 @@
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace System.IO.FileSystem.Tests
+{
+    public class File_Create_str : FileSystemTest
+    {
+        #region Utilities
+
+        public virtual FileStream Create(string path)
+        {
+            return File.Create(path);
+        }
+
+        #endregion
+
+        #region UniversalTests
+
+        [Fact]
+        public void NullPath()
+        {
+            Assert.Throws<ArgumentNullException>(() => Create(null));
+        }
+
+        [Fact]
+        public void EmptyPath()
+        {
+            Assert.Throws<ArgumentException>(() => Create(string.Empty));
+        }
+
+        [Fact]
+        public void CreateCurrentDirectory()
+        {
+            Assert.Throws<UnauthorizedAccessException>(() => Create("."));
+        }
+
+        [Fact]
+        public void ValidCreation()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string testFile = Path.Combine(testDir.FullName, GetTestFileName());
+            using (FileStream stream = Create(testFile))
+            {
+                Assert.True(File.Exists(testFile));
+                Assert.Equal(0, stream.Length);
+                Assert.Equal(0, stream.Position);
+            }
+        }
+
+        [Fact]
+        public void CreateInParentDirectory()
+        {
+            string testFile = GetTestFileName();
+            using (FileStream stream = Create(Path.Combine(TestDirectory, "DoesntExists", "..", testFile)))
+            {
+                Assert.True(File.Exists(Path.Combine(TestDirectory, testFile)));
+            }
+        }
+
+        [Fact]
+        public void LegalSymbols()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string testFile = Path.Combine(testDir.FullName, GetTestFileName() + "!@#$%^&");
+            using (FileStream stream = Create(testFile))
+            {
+                Assert.True(File.Exists(testFile));
+            }
+        }
+
+        [Fact]
+        public void InvalidDirectory()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string testFile = Path.Combine(testDir.FullName, GetTestFileName(), GetTestFileName());
+            Assert.Throws<DirectoryNotFoundException>(() => Create(testFile));
+        }
+
+        [Fact]
+        public void ValidWhitespaceNames()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string testFile1 = Path.Combine(testDir.FullName, GetTestFileName());
+            string testFile3 = Path.Combine(" ", testDir.FullName, " ", GetTestFileName());
+            using (Create(string.Format(" {0}", testFile1)))
+            using (Create(testFile3))
+            {
+                Assert.True(File.Exists(testFile1));
```

Comment:
Okay, now everything is failing properly :) 

This method still has value to test that extra whitespace in a filename is removed before file creation, but it should be marked as Windows only and a similar Unix version should be added. I'm also planning now to add a similar test to GetFileSystemEntries, as I don't think I'm currently testing it there either.